core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -239,19 +239,24 @@ class BrokerConfigHandler(private val brokerConfig: KafkaConfig,
 
   def processConfigChanges(brokerId: String, properties: Properties): Unit = {
     def getOrDefault(prop: String): Long = {
-      if (properties.containsKey(prop))
-        properties.getProperty(prop).toLong
-      else
-        QuotaConfigs.QUOTA_BYTES_PER_SECOND_DEFAULT
+      brokerConfig.dynamicConfig.currentDynamicBrokerConfigs get prop match {
```

Comment:
Maybe we don't need to update quota if `properties` has no related configs. The default value `QUOTA_BYTES_PER_SECOND_DEFAULT` is equal to "no quota limit". WDYT?

Reply:
Do you mean do something similar to the `TopicConfigHandler` `processConfigChanges` method where the update is done in the helper function? In this case we could call `updateQuota` if the value or defaultValue has changed.

Reply:
> In this case we could call updateQuota only if the value or defaultValue has changed, which could be nice.

I prefer this approach.

Reply:
I've updated it, WDYT? Sorry if my code is dodgy, this is my first time writing Scala.

Reply:
Sorry that we need `QuotaConfigs.QUOTA_BYTES_PER_SECOND_DEFAULT`, since users may remove both "dynamic" value and "default" value. In that scenario, we need to set `QuotaConfigs.QUOTA_BYTES_PER_SECOND_DEFAULT` to quota. Please ignore my previous comment.

BTW, both `currentDynamicDefaultConfigs` and `currentDynamicBrokerConfigs` do the deep copy. Could we reuse them?

Reply:
Ok, I have reverted my previous change and removed the duplicated deep copies.